### PR TITLE
Cloudfoundry - Route weight

### PR DESCRIPTION
**Pull request by nhsieh**
update virtual service for weighted-routing

### CODE REVIEW

**Comment by nhsieh** [comment]
I signed it!

**Comment by kauana** [comment]
I signed it!

File: pilot/pkg/config/cloudfoundry/controller.go [hunk 1]
```diff
@@ -279,21 +283,24 @@ func createVirtualService(gatewayNames []string, route *copilotapi.RouteWithBack
...
-					Port: &networking.PortSelector{
-						Port: &networking.PortSelector_Number{
-							Number: 8080,
-						},
-					},
+func createDestinationWeight(route *copilotapi.RouteWithBackends) *networking.DestinationWeight {
+	return &networking.DestinationWeight{
+		Destination: &networking.Destination{
+			Host:   route.GetHostname(),
+			Subset: route.GetCapiProcessGuid(),
```
Comment (rshriram) [inline pilot/pkg/config/cloudfoundry/controller.go]: where is the destination rule for this?

Comment (kauana) [reply pilot/pkg/config/cloudfoundry/controller.go]: Hello rshriram,

We did not change anything in Destination Rules, we basically updated Virtual Services to contain all destinations in the same route array. Could you please clarify what are you looking for?

Thank you.

Comment (rshriram) [reply pilot/pkg/config/cloudfoundry/controller.go]: If you specify subsets in the destination weight, you need a destinationRule that defines the subset. I don't see a destination rule in this code. I see just the virtualService. Put another way, if you define this config, you will see 503s. Is the code to generate destination rules, already present? If so, please point me to it

Comment (nhsieh) [reply pilot/pkg/config/cloudfoundry/controller.go]: The destinationRule and subsets are defined here. https://github.com/istio/istio/blob/bd17e46e3ad4151c41427aa37c8717c55b1bf360/pilot/pkg/config/cloudfoundry/controller.go#L222-L227. The destination rule and virtual services are linked through hostnames.

**Pull request merged**